runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -637,6 +640,9 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 			void* destData = getDataPointerForContiguous(destObject);
 			switch(elementSize)
 			{
+			case 0:
+				break;
+
```

Comment:
@TobiAjila This function is used for Primitive arrays only and supports element sizes 1, 2, 4, 8 bytes. Can Flattened array be declared Primitive and have different element size (except 0-stride we are adding now)? If so, what size of chunk we should use to copy? We can postpone the implementation and add it in different PR. Currently this function triggers assertion if size of the element is not supported.

Reply:
If a flattened array is declared as primitive it will only have size 1, 2, 4  or 8.

Reply:
Long term, it would be nice if we can deal with zero payload arrays. Where we have a header and no body.